pkg/broker/apiserver.go

```diff
@@ -25,22 +27,30 @@ type brokerServer struct {
 }
 
 func NewBrokerServer(addr string, giteaClient gitea.Client, mgr manager.Manager) *brokerServer { // nolint: golint
-	h := brokerapi.New(
-		&giteaServiceBroker{
-			Client:      mgr.GetClient(),
-			GiteaClient: giteaClient,
-		},
-		lager.NewZapAdapter("gitea-service-broker", zap.L()),
-		brokerapi.BrokerCredentials{
-			Username: options.Username,
-			Password: options.Password,
-		},
-	)
-	h.(*mux.Router).Use(loggingMiddleware)
+	r := mux.NewRouter()
+
+	s := r.PathPrefix("/").Subrouter()
+	serviceBroker := &giteaServiceBroker{
+		Client:      mgr.GetClient(),
+		GiteaClient: giteaClient,
+	}
+	logger := lager.NewZapAdapter("gitea-service-broker", zap.L())
+	brokerapi.AttachRoutes(s, serviceBroker, logger)
+	brokerCredentials := brokerapi.BrokerCredentials{
+		Username: options.Username,
+		Password: options.Password,
+	}
+	authMiddleware := auth.NewWrapper(brokerCredentials.Username, brokerCredentials.Password).Wrap
+	s.Use(authMiddleware)
+	s.Use(originating_identity_header.AddToContext)
+	r.Path("/healthz").Methods("GET").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Just a litte code-style nit: Move `r.Path` and `r.Use` above subrouter definition. Add a comment to `r.Path` and `s := r.PathPrefix("/").Subrouter` about their purpose.